server/aws/mc-lambda.tf

```diff
@@ -0,0 +1,113 @@
+##
+#  Mectrics Collection Lambda
+##
+
+resource "aws_lambda_function" "metrics" {
+  function_name = var.service_name
+  description   = var.lambda-description
+  #  filename      = "/tmp/lambda_validate_deploy.rb.zip"
+  s3_bucket = var.lambda_code
+  s3_key    = var.lambda-function-code
```

Comment:
As mentioned in #75 this bucket doesn't exist, please look at how this is implemented in https://github.com/cds-snc/covid-alert-server-staging-terraform/blob/master/server/aws/codedeploy.tf

Specifically the following lines (30-53): 

```terraform
data "archive_file" "lambda_validate_deploy" {
  type        = "zip"
  source_file = "lambda/lambda_validate_deploy.rb"
  output_path = "/tmp/lambda_validate_deploy.rb.zip"
}

resource "aws_lambda_function" "retrieval_lambda_validate_deploy" {
  filename      = "/tmp/lambda_validate_deploy.rb.zip"
  function_name = "RetrievalValidateBeforeTraffic"
  role          = aws_iam_role.lambda_validate_deploy.arn
  handler       = "lambda_validate_deploy.handler"

  source_code_hash = data.archive_file.lambda_validate_deploy.output_base64sha256

  runtime = "ruby2.7"

  environment {
    variables = {
      TEST_LB_ENDPOINT         = "https://${aws_lb.covidshield_key_retrieval.dns_name}:8443/services/ping"
      CLOUDFRONT_CUSTOM_HEADER = var.cloudfront_custom_header
    }
  }

}
```

Reply:
you'll need to either commit the node_modules to that folder or run npm ci in that folder before the tf action kicks off. 

alternatively you can also just create the zip package drop it in the repo and reference that without having to use the `archive_file` data resource